### PR TITLE
feat(zeebe): document new experiment job push feature

### PR DESCRIPTION
**Pull request by npepinpe**
## Description

This PR documents the new experiment push based job activation pipeline, specifically from the worker point of view.

The following was added to the Java Client API.

1. Short introduction and warning about the experimental nature of the feature
2. How to use the feature
3. Caveats

Technical details about the feature were added to the concept page of the job workers:

1. Short introduction and warning about the experimental nature of the feature
2. A high level overview of how it works
3. Troubleshooting

It is very possible I went into too much details, to be honest. I did think about separating this into a more "reference" section, but I wasn't sure where that should go, since it's still quite specific to how jobs are activated/job workers function.

I also have terrible graphics skills, so any improvements/suggestions there is welcome.

## When should this change go live?

- [ ] This change is not yet live and should not be merged until {ADD_DATE} (apply `hold` label or convert to draft PR)?
- [x] There is no urgency with this change.
- [ ] This change or page is part of a marketing blog, conference talk, or something else on a schedule.
- [ ] This functionality is already available but undocumented.
- [ ] This is a bug fix or security concern.

## PR Checklist

- [ ] I have added changes to the relevant `/versioned_docs` directory, or they are not for an **already released version**.
- [x] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

**Review by deepthidevaki — APPROVED** [review]
LGTM.

I suggest to keep the following sections in java-client/job-worker.md : 
Intro, Usage, Backfilling, Caveats.

and keep the following sections in components/concepts/job-workers: How it works

I'm not sure where to put Troubleshooting. If there is no other place, then it is better to keep it in components/concepts/job-workers.

**Comment by npepinpe** [comment]
:+1: Thanks, I applied your feedback and moved the more technical stuff to the concept page.

**Comment by akeller** [comment]
Will this be part of the 8.3 release? Are you ready for a technical writing review to prep for merging, or are you still working?

**Comment by npepinpe** [comment]
yes and yes for technical review

File: docs/apis-tools/java-client/job-worker.md [hunk 1]
```diff
@@ -41,3 +41,55 @@ By backing off, the job worker helps Zeebe by reducing the load.
...
 Zeebe's [backpressure mechanism](../../../self-managed/zeebe-deployment/operations/backpressure) can also be configured.
 :::
+
+## Job streaming
+
+:::warning
+Job streaming is an experimental feature which is still under development. It is an opt-in feature which is disabled by default.
+:::
+
+Job workers are designed to regularly poll and activate jobs. It's also possible to use them in a streaming fashion, such that jobs are automatically activated and pushed downstream to workers without requiring an extra round of polling. This greatly cuts down on overall activation latency by completely obviating the poll request.
```
Comment (christinaausley) [inline docs/apis-tools/java-client/job-worker.md:51]: Is obviating a common term across engineering? If not, I think we could substitute this for something more familiar, like "removing".

**Review by christinaausley — DISMISSED** [review]
Technical review in latest commit and one comment for clarity 👍

**Comment by christinaausley** [comment]
Let me know if you need any assistance closing this out @npepinpe 👍

**Comment by npepinpe** [comment]
Had to rebase after merging the job worker metrics PR :+1:

**Comment by npepinpe** [comment]
I suppose I need a new review in order to merge? :smile:

**Review by christinaausley — APPROVED** [review]

**Pull request merged**